### ISSUE TITLE
Set Quickery-defined attributes as READ-ONLY

### ISSUE THREAD

**Issue by jrpolidario** [jrpolidario/quickery#1]
Inherently use `attr_readonly :someattribute` to prevent updating quickery-defined attributes.

Example:

```ruby
class Employee < ApplicationRecord
  belongs_to :branch

  # put this line inherently inside the Quickery gem itself
  attr_readonly :branch_company_name

  quickery do
    branch.company.name == :branch_company_name
  end
end
```

**Comment by jrpolidario** [comment]
TL:DR; 
Skipped until there is a way to bypass `attr_readonly` when assigning attributes before saving of `Employee`.

This is because Quickery uses `before_save` callback (amongst others), that assigns the new value of `:branch_company_name` above. Although it's good that developers cannot accidentally update these attributes if `attr_readonly` is set, the problem is that Quickery won't also be able to save these attributes precisely because they are `attr_readonly`. Current preliminary research yields me no good solution to bypass `attr_readonly`. I can use `update_columns` or `update_column` to directly execute the update SQL in the database ignoring any callbacks or validations, but the problem is that Quickery uses `before_save` callback and not `after_save` callback, which means that when an `Employee` record is just about to be created, I cannot `update_columns` it inside the `before_save` precisely because I cannot update it because the record does not exist yet in the database. I was thinking of changing from `before_save` to `after_save` only to found out that `RECORD.changes` inside `before_save` behaves differently now (at Rails 5.2) inside `after_save`--[more info here](https://www.ombulabs.com/blog/rails/upgrades/active-record-5-1-api-changes.html)--which is why I'll have to write specific codes/implementation just to support both Rails 5 and 4, and probably also for future versions. This is why, I decided to skip this feature enhancement for now.
